src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,108 @@
+package parser;
+
+import com.google.gson.JsonElement;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Test
+public class JsonParserTest {
+
+    private static final double BOOK_PRICE = 25.2;
+    private static final double SOFT_PRICE = 99.0;
+    private static final String PATH_TO_CART_LIST = "/Users/dmitrymandrik/Documents/training/testng-maven/src/test/java/datasets/carts.txt";
+    private static final String PATH_TO_CART = "/Users/dmitrymandrik/Documents/training/testng-maven/src/main/resources/john-cart.json";
+    private Cart cart;
+    private JsonParser parser;
+
+    @DataProvider
+    private Object[][] setCartFilesList() {
+        BufferedReader in;
+        List<String> cartFilesList = new ArrayList<>();
+
+        try {
+            in = new BufferedReader(new FileReader(PATH_TO_CART_LIST));
+            String str;
+
+            while ((str = in.readLine()) != null) {
+                cartFilesList.add(str);
+            }
+            in.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        Object[][] returnValue = new Object[cartFilesList.size()][1];
```

Comment:
It makes sense to update this part of code to work with > 1 parameter for each dataset. Example: src/main/resources/vasya-cart.json,true
src/main/resources/fedya-cart.json,false
src/main/resources/nikitka-cart.json,true
src/main/resources/leopold-cart.json,true
src/main/resources/anna-cart.json,false
It means that 1 dataset can have > 1 parameter and it will be nice that your code will be flexible for such cases too.
If you have any questions feel free to ask me.

Reply:
Done)